### PR TITLE
[ISSUE #8153]Fix AutoExpireCache NPE

### LINKED ISSUE #8153: AutoExpireCache NPE
**Describe the bug**
This program has been running 15 days, and the NPE problem has occurred, and only once.

**How to Reproduce**
Sorry，I do not know

 - OS: [Centos]
 - Version [nacos-server 2.0.4, nacos-client 2.0.4]
 - Module [cache]
 - SDK [spring-cloud-alibaba-nacos, dubbo]

image

error message:create limit fail 
```
java.lang.NullPointerException: null 
at com.alibaba.nacos.common.cache.decorators.AutoExpireCache.isExpire(AutoExpireCache.java:90) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.common.cache.decorators.AutoExpireCache.get(AutoExpireCache.java:51) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.common.cache.decorators.AutoExpireCache.get(AutoExpireCache.java:61) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.common.cache.decorators.SynchronizedCache.get(SynchronizedCache.java:48) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.impl.Limiter.isLimit(Limiter.java:76) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.impl.ClientWorker$ConfigRpcTransportClient.requestProxy(ClientWorker.java:1002) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.impl.ClientWorker$ConfigRpcTransportClient.queryConfig(ClientWorker.java:947) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.impl.ClientWorker.getServerConfig(ClientWorker.java:397) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.NacosConfigService.getConfigInner(NacosConfigService.java:166) ~[nacos-client-2.0.4.jar:?] 
at com.alibaba.nacos.client.config.NacosConfigService.getConfig(NacosConfigService.java:94) ~[nacos-client-2.0.4.jar:?] 
at org.apache.dubbo.configcenter.support.nacos.NacosConfigServiceWrapper.getConfig(NacosConfigServiceWrapper.java:53) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.configcenter.support.nacos.NacosDynamicConfiguration.getInternalProperty(NacosDynamicConfiguration.java:234) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.configcenter.wrapper.CompositeDynamicConfiguration.lambda$getInternalProperty$4(CompositeDynamicConfiguration.java:74) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.configcenter.wrapper.CompositeDynamicConfiguration.iterateConfigOperation(CompositeDynamicConfiguration.java:115) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.configcenter.wrapper.CompositeDynamicConfiguration.getInternalProperty(CompositeDynamicConfiguration.java:74) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:124) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:111) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.CompositeConfiguration.getInternalProperty(CompositeConfiguration.java:78) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:124) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:111) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.convert(Configuration.java:144) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.Configuration.getString(Configuration.java:48) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.ConfigurationUtils.getDynamicProperty(ConfigurationUtils.java:149) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.ConfigurationUtils.lambda$getCachedDynamicProperty$0(ConfigurationUtils.java:133) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.ConfigurationCache.computeIfAbsent(ConfigurationCache.java:43) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.ConfigurationUtils.getCachedDynamicProperty(ConfigurationUtils.java:133) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.common.config.ConfigurationUtils.getCachedDynamicProperty(ConfigurationUtils.java:380) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.client.migration.DefaultMigrationAddressComparator.shouldMigrate(DefaultMigrationAddressComparator.java:71) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.client.migration.MigrationInvoker.lambda$calcPreferredInvoker$4(MigrationInvoker.java:471) ~[dubbo-3.0.5.jar:3.0.5] 
at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:1.8.0_251] 
at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1812) ~[?:1.8.0_251] 
at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126) ~[?:1.8.0_251] 
at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499) ~[?:1.8.0_251] 
at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486) ~[?:1.8.0_251] 
at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_251] 
at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:1.8.0_251] 
at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:1.8.0_251] 
at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_251] 
at java.util.stream.ReferencePipeline.allMatch(ReferencePipeline.java:454) ~[?:1.8.0_251] 
at org.apache.dubbo.registry.client.migration.MigrationInvoker.calcPreferredInvoker(MigrationInvoker.java:471) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.client.migration.MigrationInvoker.lambda$refreshInterfaceInvoker$3(MigrationInvoker.java:457) ~[dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.integration.DynamicDirectory.invokersChanged(DynamicDirectory.java:332) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.integration.RegistryDirectory.refreshInvoker(RegistryDirectory.java:266) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.integration.RegistryDirectory.refreshOverrideAndInvoker(RegistryDirectory.java:184) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.integration.RegistryDirectory.notify(RegistryDirectory.java:162) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.support.AbstractRegistry.notify(AbstractRegistry.java:440) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.support.FailbackRegistry.doNotify(FailbackRegistry.java:371) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.support.FailbackRegistry.notify(FailbackRegistry.java:363) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.nacos.NacosRegistry.notifySubscriber(NacosRegistry.java:556) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.nacos.NacosRegistry.access$100(NacosRegistry.java:88) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.nacos.NacosRegistry$RegistryChildListenerImpl$1.doNotify(NacosRegistry.java:656) [dubbo-3.0.5.jar:3.0.5] 
at org.apache.dubbo.registry.RegistryNotifier$NotificationTask.run(RegistryNotifier.java:100) [dubbo-3.0.5.jar:3.0.5] 
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_251] 
at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_251] 
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_251] 
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_251] 
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_251] 
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_251] 
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_251]
```

### PR DESCRIPTION
**Pull request by Oliverwqcwrw**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Close #8153 

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by Oliverwqcwrw** [comment]
@onewe 如果有时间的话帮忙check一下

File: common/src/main/java/com/alibaba/nacos/common/cache/decorators/AutoExpireCache.java [hunk 1]
```diff
@@ -47,7 +48,7 @@ public void put(K key, V val) {
     }
     
     @Override
-    public V get(K key) {
+    public synchronized V get(K key) {
```
Comment (onewe) [inline common/src/main/java/com/alibaba/nacos/common/cache/decorators/AutoExpireCache.java:51]: 我认为加锁并不能解决问题,NPE是发生在`isExpire`方法里面,加了锁任然会有这个问题

**Review by onewe — APPROVED** [review]

**Review by KomachiSion — CHANGES_REQUESTED** [review]
should `itemProperties==null` return true? which mean the item is expired

**Comment by KomachiSion** [comment]
Is there has unit test?

**Comment by Oliverwqcwrw** [comment]
> Is there unit test?

This problem is a low probability event under multi-threading, and the feeling plus UT may not be able to hit this problem

**Review by KomachiSion — DISMISSED** [review]

**Comment by KomachiSion** [comment]
> > Is there unit test?
> 
> This problem is a low probability event under multi-threading, and the feeling plus UT may not be able to hit this problem

Add unit test is mean your changes is expected, and can check others' change

**Comment by Oliverwqcwrw** [comment]
> > > Is there unit test?
> 
> > 
> 
> > This problem is a low probability event under multi-threading, and the feeling plus UT may not be able to hit this problem
> 
> 
> 
> Add unit test is mean your changes is expected, and can check others' change 

done

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**